File: leetcode_java/src/easy_problems/RomanToInteger.java
package easy_problems;

import java.util.HashMap;
import java.util.Map;

public class RomanToInteger {

  public static void main(String [] args) {
    RomanToInteger obj = new RomanToInteger();
    System.out.println(obj.romanToInt("XLVII"));
  }

  static Map<Character, Integer> romansMap = new HashMap<>();
  static {
    romansMap.put('I', 1);
    romansMap.put('V', 5);
    romansMap.put('X', 10);
    romansMap.put('L', 50);
    romansMap.put('C', 100);
    romansMap.put('D', 500);
    romansMap.put('M', 1000);
  }

  public int romanToInt(String s) {
    int counter = 0;
    int total = 0;
    int romanLength = s.length();

    while (counter < romanLength) {
      if (romanLength - counter >= 2 && (romansMap.get(s.charAt(counter)) < romansMap.get(s.charAt(counter + 1)))) {
        total = total + (romansMap.get(s.charAt(counter + 1)) - romansMap.get(s.charAt(counter)));
        counter += 2;
      } else {
        total = total + romansMap.get(s.charAt(counter));
        counter ++;
      }
    }
    return total;
  }

}

File: leet_code_kotlin/src/easy_problems/RomanToInteger.kt
package easy_problems

import java.util.HashMap

class RomanToInteger {

    fun romanToInt(s: String): Int {
        val romansMap: HashMap<String, Int> = hashMapOf(
            "I" to 1,
            "V" to 5,
            "X" to 10,
            "L" to 50,
            "C" to 100,
            "D" to 500,
            "M" to 1000
        )

        var result = romansMap.get(s[s.length - 1].toString())!!
        for(i in s.length -2 downTo 0) {
            romansMap.get(s[i].toString())?.let{current ->
                romansMap.get(s[i + 1].toString())?.let{after ->
                    if (current < after) {
                        result -= current
                    } else {
                        result += current
                    }
                }
            }
        }
        return result
    }

    fun main(args: Array<String>) {
        val obj = RomanToInteger()
        println(message = obj.romanToInt("XLVII"))
    }
}